src/experimental/reputation/system.ts

```diff
@@ -0,0 +1,233 @@
+import { Proposal, ProposalFilter } from "../../market";
+import { AgreementSelector } from "../../agreement";
+import { GolemReputationError } from "./error";
+import {
+  AgreementSelectorOption,
+  ProposalFilterOptions,
+  ReputationConfig,
+  ReputationData,
+  ReputationProviderEntry,
+  ReputationProviderScores,
+  ReputationWeights,
+} from "./types";
+import { Logger, nullLogger } from "../../utils";
+
+/**
+ * Default minimum score for proposals.
+ * @experimental
+ */
+export const DEFAULT_PROPOSAL_MIN_SCORE = 0.8;
+
+/**
+ * Default weights used to calculate the score for proposals.
+ * @experimental
+ */
+export const DEFAULT_PROPOSAL_WEIGHTS: ReputationWeights = {
+  uptime: 0.5,
+  successRate: 0.5,
+} as const;
+
+/**
+ * Default weights used to calculate the score for agreements.
+ * @experimental
+ */
+export const DEFAULT_AGREEMENT_WEIGHTS: ReputationWeights = {
+  uptime: 0.5,
+  successRate: 0.5,
+} as const;
+
+/**
+ * Default reputation service URL.
+ * @experimental
+ */
+export const DEFAULT_REPUTATION_URL = "https://asia.reputation.dev-test.golem.network/v1/providers/scores";
+
+/**
+ * Reputation system client.
+ *
+ * This class is responsible for fetching and applying reputation data to Golem SDK's market management class.
+ *
+ * Currently, it includes a proposal filter you can use to filter out providers with low reputation scores.
+ *
+ * Reputation data is gathered by the following project: https://github.com/golemfactory/reputation-auditor
+ *
+ * You can adjust the weights used to calculate the score for proposals by using the `setProposalWeights` method.
+ *
+ * NOTE: This class is currently experimental and subject to change.
+ *
+ * NOTE: Only providers from polygon network are being tested, so using this class on testnet will not work.
```

Comment:
Worse, it will throw an error so we could point this out in the docs explicitly :)

I was running into the same issue with my golem-node-validator and whitelisting solution. There I took the alternative approach

> Query the whitelist with the `paymentNetwork` parameter. If the returned data set is empty, act like an "accept all" filter and allow the proposal to be processed by other filters set by the user.

The reason why I did that is that once upon a time I introduced a bug in my validator that led to an empty whitelist over time. What I mean by that is: if you throw an error to the user when the data set is empty or you have a network issue with reaching that endpoint you essentially take responsibility for the uptime of the user's application. If the reputation endpoint is going to become unavailable, so will the user applications.

Just before we continue - is this the behaviour your're willing to introduce?